include/syncmer_hash.hpp

```diff
@@ -0,0 +1,139 @@
+    * \returns               A range of converted elements.
+    */
+    constexpr auto operator()(size_t const smers, size_t const kmers, seed const seed) const
+    {
+        return seqan3::detail::adaptor_from_functor{*this, smers, kmers, seed};
+    }
+
+    /*!\brief Call the view's constructor with the underlying view, a window size and a subwindow size as argument.
+     * \param[in] urange      The input range to process. Must model std::ranges::viewable_range and
+     *                        the reference type of the range must model seqan3::semialphabet.
+     * \param[in] kmers       The K-mer size to be used.
+     * \param[in] smers       The S-mer size (s<k) to be used.
+     * \param[in] seed        The seed to use.
+     * \throws std::invalid_argument if the subwindow size is smaller than 1 or kmers is smaller than smers.
+     * \returns               A range of converted elements.
+     */
+    template <std::ranges::range urng_t>
+    constexpr auto operator()(urng_t && urange,
+                              size_t const smers,
+			      size_t const kmers,
+                              seed const seed = seqan3::seed{0x8F3F73B5CF1C9ADE}) const
+    {
+        static_assert(std::ranges::viewable_range<urng_t>,
+            "The range parameter to views::syncmer_hash cannot be a temporary of a non-view range.");
+        static_assert(std::ranges::forward_range<urng_t>,
+            "The range parameter to views::syncmer_hash must model std::ranges::forward_range.");
+        static_assert(semialphabet<std::ranges::range_reference_t<urng_t>>,
+            "The range parameter to views::syncmer_hash must be over elements of seqan3::semialphabet.");
+
+        if (smers < 1 || kmers <= smers)
+            throw std::invalid_argument{"The chosen kmers and smers are not valid."
+                                        "Please choose values greater than 1 and a smer size smaller than the kmer size."};
+
+        auto kmers_hash = std::forward<urng_t>(urange)
+                                                 | seqan3::views::kmer_hash(seqan3::shape(seqan3::ungapped(kmers)))
+                                                 | std::views::transform([seed] (uint64_t i)
+                                                          {return i ^ seed.get();});
+
+        auto smers_hash = std::forward<urng_t>(urange)
+                                                 | seqan3::views::kmer_hash(seqan3::shape(seqan3::ungapped(smers)))
+                                                 | std::views::transform([seed] (uint64_t i)
+                                                          {return i ^ seed.get();});
+
+        return seqan3::detail::syncmer_view(smers_hash, kmers_hash, kmers - smers + 1);
+    }
+};
+
+} // namespace seqan3::detail
+
+/*!\name Alphabet related views
+ * \{
+ */
+
+/*!\brief                     Computes syncmers for a range with given window and subwindow sizes, and seed.
```

Comment:
Here you can actually use the words k-mers and s-mers, because that is the variable names.